compiler/rustc_resolve/src/macros.rs

```diff
@@ -769,6 +769,15 @@ impl<'a> Resolver<'a> {
                     let expected = kind.descr_expected();
                     let msg = format!("cannot find {} `{}` in this scope", expected, ident);
                     let mut err = self.session.struct_span_err(ident.span, &msg);
+                    if ident.to_string() == "default"
+                        && let ModuleKind::Def(DefKind::Enum, def_id, _) = parent_scope.module.kind
+                        && let Some(span) = self.opt_span(def_id)
+                    {
+                        err.span_help(
+                            self.session.source_map().guess_head_span(span),
+                            "consider adding `#[derive(Default)]` to this enum"
+                        );
```

Comment:
This should ideally be `span_suggestion` using `span_to_snippet` on the source map.